src/components/profile.ts

```diff
@@ -0,0 +1,153 @@
+import { openDB } from './db';
+
+// since interfaces are for compile time there is no easy way to extract keys
+// so this userProfileKeys is needed
+export const userProfileKeys = [
+  'about_me',
+  'birth_date',
+  'preferred_name',
+  'preferred_pronouns',
+  'term',
+  'year',
+  'faculty',
+  'program',
+  'specialization'
+];
+
+export interface UserProfile {
+  about_me?: string;
+  birth_date?: string;
+  preferred_name?: string;
+  preferred_pronouns?: string;
+  term?: string;
+  year?: number;
+  faculty?: string;
+  program?: string;
+  specialization?: string;
+}
+
+const validFaculties = ['Mathematics', 'Engineering', 'Arts', 'Health', 'Science'];
+const validTerms: string[] = ['1A', '1B', '2A', '2B', '3A', '3B', '4A', '4B', 'MASc', 'PHD'];
+const validYears: number[] = [];
+const yearStart = 2000;
```

Comment:
hmm maybe we shouldn't restrict it to just 2000-2030
and use curr year +1 as a max
and u never know if a 1999 grad comes on disc

Reply:
curr year + 6 probs, since 2022+1 = 2023, therefore I can't even put my year